src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -8,111 +8,110 @@
 
-            return queryBuilder;
+            _queryBuilders.TryAdd(dataSource.DatabaseType, queryBuilder!);
+            _queryExecutors.TryAdd(dataSource.DatabaseType, queryExecutor!);
+            _dbExceptionsParsers.TryAdd(dataSource.DatabaseType, exceptionParser!);
         }
+    }
 
-        /// <inheritdoc />
-        public IQueryExecutor GetQueryExecutor(DatabaseType databaseType)
+    /// <inheritdoc />
+    public IQueryBuilder GetQueryBuilder(DatabaseType databaseType)
+    {
+        if (!_queryBuilders.TryGetValue(databaseType, out IQueryBuilder? queryBuilder))
         {
-            if (!_queryExecutors.TryGetValue(databaseType, out IQueryExecutor? queryExecutor))
-            {
-                throw new DataApiBuilderException(
-                    $"{nameof(databaseType)}:{databaseType} could not be found within the config",
-                    HttpStatusCode.BadRequest,
-                    DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
-            }
-
-            return queryExecutor;
+            throw new DataApiBuilderException(
+                $"{nameof(DatabaseType)}:{databaseType} could not be found within the config",
+                HttpStatusCode.BadRequest,
+                DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
         }
 
-        /// <inheritdoc />
-        public DbExceptionParser GetDbExceptionParser(DatabaseType databaseType)
+        return queryBuilder;
+    }
+
+    /// <inheritdoc />
+    public IQueryExecutor GetQueryExecutor(DatabaseType databaseType)
+    {
+        if (!_queryExecutors.TryGetValue(databaseType, out IQueryExecutor? queryExecutor))
         {
-            if (!_dbExceptionsParsers.TryGetValue(databaseType, out DbExceptionParser? exceptionParser))
-            {
-                throw new DataApiBuilderException(
-                    $"{nameof(databaseType)}:{databaseType} could not be found within the config",
-                    HttpStatusCode.BadRequest,
-                    DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
-            }
+            throw new DataApiBuilderException(
+                $"{nameof(databaseType)}:{databaseType} could not be found within the config",
+                HttpStatusCode.BadRequest,
+                DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
+        }
+
+        return queryExecutor;
+    }
 
-            return exceptionParser;
+    /// <inheritdoc />
+    public DbExceptionParser GetDbExceptionParser(DatabaseType databaseType)
+    {
+        if (!_dbExceptionsParsers.TryGetValue(databaseType, out DbExceptionParser? exceptionParser))
+        {
+            throw new DataApiBuilderException(
+                $"{nameof(databaseType)}:{databaseType} could not be found within the config",
+                HttpStatusCode.BadRequest,
+                DataApiBuilderException.SubStatusCodes.DataSourceNotFound);
         }
 
+        return exceptionParser;
     }
+
```

Comment:
nit: I know you didnt add this empty line, but for consistency with the other files we should remove it.

Reply:
can resolve in follow up.